Mathlib/Algebra/Group/Subgroup/MulOpposite.lean

```diff
@@ -86,17 +85,41 @@ def opEquiv : Subgroup G ≃o Subgroup Gᵐᵒᵖ where
   right_inv := op_unop
   map_rel_iff' := op_le_op_iff
 
+@[to_additive]
+theorem op_injective : (@Subgroup.op G _).Injective := opEquiv.injective
```

Comment:
Can you add all these new lemmas for `Submonoid` and `Subsemiring` and `Subring` too?

Reply:
@eric-wieser Done